### PR TITLE
 Add support for Chromium OS EC devices

### PR DESCRIPTION
**Pull request by fandango96**
This PR adds the `CROS_EC_DEV_IOCXCMD_V2` and `CROS_EC_DEV_IOCEVENTMASK_V2` `ioctl`s necessary for testing [Chromium OS EC devices](https://chromium.googlesource.com/chromiumos/platform/ec/+/HEAD/README.md). My main use-case is to add support for the FPMCU to allow automated testing in `libfprint`.

PR showing use of the newly added `ioctl`s here: https://gitlab.freedesktop.org/libfprint/libfprint/-/merge_requests/493

### CODE REVIEW

**Comment by fandango96** [comment]
not sure why the build fails only on s390x

**Comment by martinpitt** [comment]
Thanks @fandango96 ! Sorry, I don't have much time this week for a detailed review, as I'm travelling. But for s390 the [build log](https://download.copr.fedorainfracloud.org/results/packit/martinpitt-umockdev-249/fedora-rawhide-s390x/07581093-umockdev/builder-live.log.gz) shows a unit test failure:
```
# DEBUG: umockdev-ioctl.vala:904: Aborting client because ioctl tree for /dev/cros_fp is empty
stderr:
ioctl_varlenstruct_init_from_text: ioctl C014EC00: expected data length 67108884, but got 24 bytes from text data
ERROR: libumockdev-preload: Server requested abort on device /dev/cros_fp, exiting
â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•â€•
```
s390x is big endian, this smells like reading a 32 bit int the wrong way around?

**Comment by fandango96** [comment]
Thank you @martinpitt, that was it - fixed.

File: meson.build [hunk 1]
```diff
@@ -135,7 +135,7 @@ preload_lib = shared_library('umockdev-preload',
    'src/ioctl_tree.c'],
   c_args: ['-fvisibility=default'],
   version: '0.0.0',
-  dependencies: [dl, pthread],
+  dependencies: [dl, glib, pthread],
```
Comment (martinpitt) [inline meson.build:138]: Please no -- this is a way too heavy and unpredictable dependency for the preload library.

File: src/ioctl_tree.c [hunk 2]
```diff
@@ -910,7 +908,7 @@ cros_ec_ioctl_get_data_size(IOCTL_REQUEST_TYPE _id, const void *data)
 {
     const struct cros_ec_command_v2 *s_cmd = (struct cros_ec_command_v2 *)data;
 
-    return sizeof(struct cros_ec_command_v2) + s_cmd->insize;
+    return sizeof(struct cros_ec_command_v2) + GUINT32_FROM_LE(s_cmd->insize);
```
Comment (martinpitt) [inline src/ioctl_tree.c:911]: Please use `le32toh()` from the standard C library here instead, see endian(3).

**Review by martinpitt — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (fandango96) [reply meson.build]: removed, hadn't used glib before my `libfprint` PR so was unaware of the cost

**Review by martinpitt — APPROVED** [review]
This looks fine, thanks! Do you need this in a new release, or is git main enough for the libfprint MR?

**Comment by fandango96** [comment]
> This looks fine, thanks! Do you need this in a new release, or is git main enough for the libfprint MR?

Thanks! git main is enough for the libfprint MR

**Comment by fandango96** [comment]
> This looks fine, thanks! Do you need this in a new release, or is git main enough for the libfprint MR?

@martinpitt looks like the CI doesn't need a new release (as it uses main) but we do a version check during runtime and I'll need to update to a version including this change. Are you planning to do a new release soon?

**Comment by martinpitt** [comment]
@fandango96 Sure! I just released https://github.com/martinpitt/umockdev/releases/tag/0.18.4 and will upload to Fedora and Debian now.

**Comment by fandango96** [comment]
> @fandango96 Sure! I just released https://github.com/martinpitt/umockdev/releases/tag/0.18.4 and will upload to Fedora and Debian now.

Thank you!

**Pull request merged**